libraries/androidtools/impl/src/main/kotlin/io/element/android/libraries/androidtools/impl/AndroidFileSizeFormatter.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.androidtools.impl
+
+import android.content.Context
+import android.os.Build
+import android.text.format.Formatter
+import com.squareup.anvil.annotations.ContributesBinding
+import io.element.android.libraries.androidtools.api.FileSizeFormatter
+import io.element.android.libraries.di.AppScope
+import io.element.android.libraries.di.ApplicationContext
+import javax.inject.Inject
+
+@ContributesBinding(AppScope::class)
+class AndroidFileSizeFormatter @Inject constructor(
```

Comment:
Then why bother at all? :) Could we simply do without this piece of code?

Reply:
We can check what other Element clients are displaying, but I believe this change was done to avoid EleWeb / iOS and Android display different values for the file sizes. Also was done to avoid having a limit for file upload on homeserver and file size displayed to the user not consistent. See https://github.com/vector-im/element-android/pull/589 .